hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/io/ECKeyOutputStream.java

```diff
@@ -303,13 +327,40 @@ private StripeWriteStatus handleParityWrites(int parityCellSize,
       if (allocateBlockIfFull) {
         blockOutputStreamEntryPool.allocateBlockIfNeeded();
       }
+      currentBlockGroupLen = 0;
     } else {
       streamEntry.resetToFirstEntry();
     }
-    currentBlockGroupLen = 0;
+
     return StripeWriteStatus.SUCCESS;
   }
 
+  private boolean isDataWriteSuccess() {
```

Comment:
This two method, along with the unchanged addToExcludedNodesList method seems to belong to ECBlockOutputStreamEntryPool, as all the method logic operates on that class, or the current stream entry which is contained by the pool instance. I think these should be moved over to ECBlockOutputStreamEntryPool.

Also all of the places where these methods are called, the result is negated in the check, so we can invert the return values, and call these methods as hasWriteFailure and hasPutBlockFailure.

Reply:
I changed the method names. Even though we are operating on the ECBlockOutputStreamEntryPool data structures, the usage logic should be in output stream for better readability. Especially I don;t want adding into excludeLists logic to be hidden somewhere in other classes. Ideally ECBlockOutputStreamEntryPool should be used as pool for keeping the entries, but we added some additional logics to it.
But I agree that method names can be changed, so that we don't need to negate in if checks. Let me know if you still feel concerned on this, we can discuss and move forward on this.

Reply:
For now I can accept the current state, but on the long run, I think all this should be in the BlockOutputStreamEntry not even in the pool level, as the entry handles a blockgroup and I believe it should also handle the write encode and retry logic, as that would massively simplify some parts of the code, but that we already discussed and agreed to deal with later on.